flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/BaseExpressions.java

```diff
@@ -2046,11 +2046,72 @@ public OutType jsonQuery(
             JsonQueryWrapper wrappingBehavior,
             JsonQueryOnEmptyOrError onEmpty,
             JsonQueryOnEmptyOrError onError) {
+        return jsonQuery(path, DataTypes.STRING(), wrappingBehavior, onEmpty, onError);
+    }
+
+    /**
+     * Extracts JSON values from a JSON string.
+     *
+     * <p>This follows the ISO/IEC TR 19075-6 specification for JSON support in SQL. The result is
```

Comment:
Reduce code duplication and simplify this JavaDoc. Link to the other and focus on the data type in following paragraphs. Also the examples should show the usage of the `datatype` parameter.